tests/test_md.py

```diff
@@ -210,8 +210,10 @@ def integrate_once_through(
             test_xs,
             ref_all_xs[::x_interval]
         )
+        self.assertEqual(test_boxes.shape[0], test_xs.shape[0])
+        self.assertEqual(test_boxes.shape[1], D)
+        self.assertEqual(test_boxes.shape[2], test_xs.shape[2])
```

Comment:
Could we also add an assertion that `test_boxes[0]` and `test_boxes[-1]` are  consistent with `ctxt_2.get_box()` before and after trajectory?

Reply:
I don't think this is guaranteed since boxes, like coords, use "pre-step" saving:

``` context.cu

        for(int i=0; i < lambda_schedule.size(); i++) {
            // decide if we need to store the du_dl for this step
            unsigned long long *du_dl_ptr = nullptr;
            if(i % store_du_dl_interval == 0) {
                // pemdas but just to make it clear we're doing pointer arithmetic
                du_dl_ptr = d_du_dl_buffer + (i / store_du_dl_interval);
            }

            if(i % store_x_interval == 0) {
                gpuErrchk(cudaMemcpy(
                    &h_x_buffer[0] + (i / store_x_interval)*N_*3,
                    d_x_t_,
                    N_*3*sizeof(double),
                    cudaMemcpyDeviceToHost)
                );
            }

            double lambda = lambda_schedule[i];
            this->_step(lambda, du_dl_ptr);
        }
```

Reply:
It is guaranteed, at least in this test, as there is no barostat.